BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -49,22 +49,68 @@ public actor DefaultTabDataStore: TabDataStore {
     // MARK: Fetching Window Data
 
     public func fetchWindowData(uuid: UUID) async -> WindowData? {
-        logger.log("Attempting to fetch window/tab data", level: .debug, category: .tabs)
+        logger.log("Attempting to fetch window data", level: .debug, category: .tabs)
         do {
-            guard let fileURL = windowURLPath(for: uuid, isBackup: false),
-                  fileManager.fileExists(atPath: fileURL),
-                  let windowData = parseWindowDataFile(fromURL: fileURL) else {
-                logger.log("Failed to open window/tab data for UUID: \(uuid)", level: .fatal, category: .tabs)
+            guard let fileURL = windowURLPath(for: uuid, isBackup: false) else {
+                logger.log(
+                    "Failed to open window data for UUID: \(uuid), nil fileURL",
+                    level: .fatal,
+                    category: .tabs
+                )
                 throw TabDataError.failedToFetchData
             }
+
+            guard fileManager.fileExists(atPath: fileURL) else {
+                logger.log(
+                    "Failed to open window data for UUID: \(uuid), file doesn't exist",
+                    level: .fatal,
+                    category: .tabs
+                )
+                throw TabDataError.failedToFetchData
+            }
+
+            guard let windowData = parseWindowDataFile(fromURL: fileURL) else {
+                logger.log(
+                    "Failed to open window data for UUID: \(uuid), file parsing failed",
+                    level: .fatal,
+                    category: .tabs
+                )
+                throw TabDataError.failedToFetchData
```

Comment:
Ahh you beat me to this, I was just going to ask if it was a problem that they wouldn't all fall under one umbrella now!

I think what you've suggested makes sense. Does it work to put the additional info into the logger extra fields? Is that easy enough to parse and filter on in Sentry?

There are two parts at play with the backup file though so that might complicate the defer logging a bit. I'll see what I can do to consolidate this a bit. 🤔

Reply:
Closing, refactored as discussed in DM